mypy/server/aststrip.py

```diff
@@ -64,16 +64,20 @@ def visit_name_expr(self, node: NameExpr) -> None:
     def visit_member_expr(self, node: MemberExpr) -> None:
         self.strip_ref_expr(node)
         if self.is_duplicate_attribute_def(node):
-            # This is marked as a instance variable definition but a base class
+            # This is marked as an instance variable definition but a base class
             # defines an attribute with the same name, and we can't have
             # multiple definitions for an attribute. Defer to the base class
             # definition.
-            del self.type.names[node.name]
+            if self.type is not None:
+                del self.type.names[node.name]
             node.is_def = False
             node.def_var = None
 
     def is_duplicate_attribute_def(self, node: MemberExpr) -> bool:
-        if not node.is_def or node.name not in self.type.names:
+        if not node.is_def:
+            return False
+        assert self.type is not None, "members can't be defined outside a class"
```

Comment:
The phrasing makes it sound like this could be a user error. But in that case it shouldn't be an assert, it should be a fail() call or something like that. So I assume it can never happen unless there's a bug elsewhere in mypy, in which case it would be better if the message was "Member defined outside class" or perhaps omitted.